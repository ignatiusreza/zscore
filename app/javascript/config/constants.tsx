// prettier-ignore
export const SAMPLE_DATA = [
  1, 2, 1, 0, 1, 2, 1, 8, 9, 8,
  1, 2, 0, 2, 1, 2, 3, 1, 2, 0,
  8, 9, 2, 0, 3, 0, 2, 1, 2, 3,
  8, 10, 2, 1, 2, 3, 0, 1, 2, 1,
  2, 7, 6, 9, 1, 2, 0, 1, 2, 1,
];
