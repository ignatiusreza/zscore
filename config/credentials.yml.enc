V5m95Vq2a80ykOjS5vJXiK63tIrM21yaKMkyw1CetRcDdbV+kK9eXenNa7aDsMT9jELhjCSLj16RWwuAJqwaYJLtIiEI6i1kJLnL4K16MVcjaN6ue5c2XWGzC5OE+3rTl3jkVipvkZXi4qoiF2qyEheIpWZDQqwgWmNA01VMvdGe3HjwN93lvm9vl1LM0U4Tq5BT3jvL2CgHSfd22PKJTGIhLhnnOgwLINYkzJ5edLVPK9HKdmAvbIxMN3rlqeoL+oNKKYKMCXcpt3lOvd72wBozMquG/BkpDDK7I4c6y2x74AYTn9XSDAltHevuS/OQOGFw9lxB09vvVWQsnB4DD7vWbiOzLU2MBqXs4tjjon9zMdPEfvenpK38/JmJOFoXPxEOmzMCo/DTUcHYrR4zhr9ugZieC0CeN8sk--h8nwDt3vAE1cuLlk--fX2gmvee5RvUYL915DmnqA==