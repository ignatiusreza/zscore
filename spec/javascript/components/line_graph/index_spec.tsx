import React from 'react';
import Renderer from 'react-test-renderer';

import { SAMPLE_DATA } from 'config/constants';
import LineGraph from 'components/line_graph';

describe('<LineGraph>', () => {
  it('match snapshots for sample data', () => {
    const output = Renderer.create(<LineGraph data={SAMPLE_DATA} />).toJSON();

    expect(output).toMatchSnapshot();
  });
});
