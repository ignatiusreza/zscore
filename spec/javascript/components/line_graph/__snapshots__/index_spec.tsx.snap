// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineGraph> match snapshots for sample data 1`] = `
<div
  className="container"
>
  <div
    className="chart"
  >
    <svg
      height={130}
      viewBox="-10 0 510 120"
      width={520}
    >
      <polyline
        className="line"
        points="0,100 10,90 20,100 30,110 40,100 50,90 60,100 70,30 80,20 90,30 100,100 110,90 120,110 130,90 140,100 150,90 160,80 170,100 180,90 190,110 200,30 210,20 220,90 230,110 240,80 250,110 260,90 270,100 280,90 290,80 300,30 310,10 320,90 330,100 340,90 350,80 360,110 370,100 380,90 390,100 400,90 410,40 420,50 430,20 440,100 450,90 460,110 470,100 480,90 490,100"
      />
      <g />
      <line
        className="grid"
        x1="0"
        x2={500}
        y1={110}
        y2={110}
      />
      <g
        className="labels xLabels"
      >
        <text
          x={0}
          y={110}
        >
          0
        </text>
        <text
          x={10}
          y={110}
        >
          1
        </text>
        <text
          x={20}
          y={110}
        >
          2
        </text>
        <text
          x={30}
          y={110}
        >
          3
        </text>
        <text
          x={40}
          y={110}
        >
          4
        </text>
        <text
          x={50}
          y={110}
        >
          5
        </text>
        <text
          x={60}
          y={110}
        >
          6
        </text>
        <text
          x={70}
          y={110}
        >
          7
        </text>
        <text
          x={80}
          y={110}
        >
          8
        </text>
        <text
          x={90}
          y={110}
        >
          9
        </text>
        <text
          x={100}
          y={110}
        >
          10
        </text>
        <text
          x={110}
          y={110}
        >
          11
        </text>
        <text
          x={120}
          y={110}
        >
          12
        </text>
        <text
          x={130}
          y={110}
        >
          13
        </text>
        <text
          x={140}
          y={110}
        >
          14
        </text>
        <text
          x={150}
          y={110}
        >
          15
        </text>
        <text
          x={160}
          y={110}
        >
          16
        </text>
        <text
          x={170}
          y={110}
        >
          17
        </text>
        <text
          x={180}
          y={110}
        >
          18
        </text>
        <text
          x={190}
          y={110}
        >
          19
        </text>
        <text
          x={200}
          y={110}
        >
          20
        </text>
        <text
          x={210}
          y={110}
        >
          21
        </text>
        <text
          x={220}
          y={110}
        >
          22
        </text>
        <text
          x={230}
          y={110}
        >
          23
        </text>
        <text
          x={240}
          y={110}
        >
          24
        </text>
        <text
          x={250}
          y={110}
        >
          25
        </text>
        <text
          x={260}
          y={110}
        >
          26
        </text>
        <text
          x={270}
          y={110}
        >
          27
        </text>
        <text
          x={280}
          y={110}
        >
          28
        </text>
        <text
          x={290}
          y={110}
        >
          29
        </text>
        <text
          x={300}
          y={110}
        >
          30
        </text>
        <text
          x={310}
          y={110}
        >
          31
        </text>
        <text
          x={320}
          y={110}
        >
          32
        </text>
        <text
          x={330}
          y={110}
        >
          33
        </text>
        <text
          x={340}
          y={110}
        >
          34
        </text>
        <text
          x={350}
          y={110}
        >
          35
        </text>
        <text
          x={360}
          y={110}
        >
          36
        </text>
        <text
          x={370}
          y={110}
        >
          37
        </text>
        <text
          x={380}
          y={110}
        >
          38
        </text>
        <text
          x={390}
          y={110}
        >
          39
        </text>
        <text
          x={400}
          y={110}
        >
          40
        </text>
        <text
          x={410}
          y={110}
        >
          41
        </text>
        <text
          x={420}
          y={110}
        >
          42
        </text>
        <text
          x={430}
          y={110}
        >
          43
        </text>
        <text
          x={440}
          y={110}
        >
          44
        </text>
        <text
          x={450}
          y={110}
        >
          45
        </text>
        <text
          x={460}
          y={110}
        >
          46
        </text>
        <text
          x={470}
          y={110}
        >
          47
        </text>
        <text
          x={480}
          y={110}
        >
          48
        </text>
        <text
          x={490}
          y={110}
        >
          49
        </text>
      </g>
    </svg>
  </div>
  <div
    className="frozen"
  >
    <div
      className="whiteout"
    />
    <svg
      height={130}
      viewBox="-10 0 510 120"
      width={520}
    >
      <rect
        className="whiteout"
        height={110}
        width={10}
        x={-10}
      />
      <rect
        className="whiteout"
        height={10}
        width={6.666666666666667}
        x={-10}
        y={110}
      />
      <line
        className="grid"
        x1="0"
        x2="0"
        y1={110}
        y2="0"
      />
      <g
        className="labels yLabels"
      >
        <text
          x="0"
          y={110}
        >
          0
        </text>
        <text
          x="0"
          y={100}
        >
          1
        </text>
        <text
          x="0"
          y={90}
        >
          2
        </text>
        <text
          x="0"
          y={80}
        >
          3
        </text>
        <text
          x="0"
          y={70}
        >
          4
        </text>
        <text
          x="0"
          y={60}
        >
          5
        </text>
        <text
          x="0"
          y={50}
        >
          6
        </text>
        <text
          x="0"
          y={40}
        >
          7
        </text>
        <text
          x="0"
          y={30}
        >
          8
        </text>
        <text
          x="0"
          y={20}
        >
          9
        </text>
        <text
          x="0"
          y={10}
        >
          10
        </text>
      </g>
    </svg>
  </div>
</div>
`;
